File: CivitasRuby/lib/casilla_calle.rb
#encoding:utf-8

require_relative 'titulo_propiedad'
require_relative 'casilla'
require 'casilla.rb'

module Civitas
    class CasillaCalle < Casilla
        def initialize(titulo)
            super(titulo.nombre)
            @importe = titulo.precio_compra
            @titulo_propiedad = titulo
        end

        def recibe_jugador(actual, todos)
            if(jugador_correcto(actual,todos))
                informe(actual,todos)
                jugador=todos[actual]
                
                if(!@titulo_propiedad.tiene_propietario)
                  jugador.puede_comprar_casilla
                  jugador.puede_comprar=true
                  
                else
                  @titulo_propiedad.tramitar_alquiler(jugador)
                  jugador.puede_comprar=false
                end
            end
        end

        def to_string
            return @titulo_propiedad.to_string 
        end
    end
end

File: CivitasJava/src/civitas/CasillaCalle.java
package civitas;

import java.util.ArrayList;

public class CasillaCalle extends Casilla{
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        importe = titulo.getPrecioCompra();
        tituloPropiedad=titulo;
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            Jugador jugador=todos.get(actual);

            if(!tituloPropiedad.tienePropietario()){
                jugador.puedeComprarCasilla();
                jugador.puedeComprar=true;
            }

            else{
                tituloPropiedad.tramitarAlquiler(jugador);
                jugador.puedeComprar=false;
            }
        }
    }

    @Override
    public String toString(){
        String str=tituloPropiedad.toString();


        //str="******************";
        //str+="Calle " + getNombre() + "    |    titulo propiedad: " + tituloPropiedad;
        //str+="******************";
        
        return str;
    }

}